### PR TITLE
Fixed docs markdown generation and fixed a number of related formatting errors

### PR DESCRIPTION
**Pull request by zachmu**

### CODE REVIEW

File: go/cmd/dolt/commands/read_tables.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (cmd ReadTablesCmd) createArgParser() *argparser.ArgParser {
 		{"commit", "Branch or commit hash representing a point in time to retrieve tables from"},
 		{"table", " Optional tables to retrieve.  If omitted, all tables are retrieved."},
 	}
-	ap.SupportsString(dirParamName, "d", "directory", "directory to create and put retrieved table data.")
+	ap.SupportsString(fileParamName, "d", "directory", "directory to create and put retrieved table data.")
```
Comment (max-hoffman) [inline go/cmd/dolt/commands/read_tables.go:75]: should the flags change to `--file, -f`?

File: go/cmd/dolt/commands/reset.go [hunk 2]
```diff
@@ -36,20 +35,19 @@ const (
...
-	This means that {{.EmphasisLeft}}dolt reset <tables>{{.EmphasisRight}} is the opposite of {{.EmphasisLeft}}dolt add <tables>{{.EmphasisRight}}.
-
-	After running {{.EmphasisLeft}}dolt reset <tables>{{.EmphasisRight}} to update the staged tables, you can use {{.EmphasisLeft}}dolt checkout{{.EmphasisRight}} to check the
-	contents out of the staged tables to the working tables.
-
-dolt reset .
-	This form resets {{.EmphasisLeft}}all{{.EmphasisRight}} staged tables to their values at HEAD. It is the opposite of {{.EmphasisLeft}}dolt add .{{.EmphasisRight}}`,
-
+	LongDesc: "Sets the state of a table in the staging area to be that table's value at HEAD\n\n" +
+		"{{.EmphasisLeft}}dolt reset <tables>...{{.EmphasisRight}}" +
```
Comment (max-hoffman) [inline go/cmd/dolt/commands/reset.go:39]: kind of awkward formatting, is there a way to do with with a backtick block or something?

File: go/cmd/dolt/cli/documentation_helper.go [hunk 3]
```diff
@@ -199,31 +190,21 @@ var CliFormat = docFormat{"<", ">", "<b>", "</b>"}
 // Special format for the synopsis which is rendered inside raw HTML in markdown
 var SynopsisMarkdownFormat = docFormat{"&lt;", "&gt;", "`", "`"}
 
-func transformSynopsisToHtml(commandStr string, synopsis []string) string {
+func transformSynopsisToMarkdown(commandStr string, synopsis []string) string {
```
Comment (max-hoffman) [inline go/cmd/dolt/cli/documentation_helper.go:193]: i don't understand the context of why you're switching from HTML to markdown here, I'm assuming you know what you're doing.  afaik most of this PR just looks like `dolt --help` printing, were we gatsby formatting those before?

Comment (max-hoffman) [reply go/cmd/dolt/cli/documentation_helper.go:193]: tests would have made it cleared where this is used

**Review by max-hoffman — APPROVED** [review]
LGMT, a couple questions about things I didn't understand

File: go/cmd/dolt/cli/documentation_helper.go (hunk 3 above)
Comment (zachmu) [reply go/cmd/dolt/cli/documentation_helper.go:193]: It was always markdown, just markdown with embedded HTML

File: go/cmd/dolt/commands/read_tables.go (hunk 1 above)
Comment (zachmu) [reply go/cmd/dolt/commands/read_tables.go:75]: This is a good catch, turns out this command was reusing the same const and I broke it by renaming it

**Pull request merged**